src/main/java/graphql/execution/ExecutionStrategy.java

```diff
@@ -558,10 +557,28 @@ protected FieldValueInfo completeValueForList(ExecutionContext executionContext,
 
         return FieldValueInfo.newFieldValueInfo(LIST)
                 .fieldValue(overallResult)
-                .fieldValueInfos(fieldValueInfos)
+                .fieldValueInfos(combineFutures(fieldValueInfos))
                 .build();
     }
 
+    private <T> List<T> combineFutures(final List<CompletableFuture<T>> futures) {
+        return futures
+                .stream()
+                .map(CompletableFuture::join)
+                .collect(Collectors.toList());
+    }
```

Comment:
This is the opposite of async - here your are forcing a synchronisation point on all the CFs at this point.

See `java.util.concurrent.CompletableFuture#allOf`